### DISCUSSION TITLE
Accessing SQL Server Container in mac

### DISCUSSION THREAD

**Discussion by Hypothesys-Smithgor** [microsoft/aspire#6269] [Q&A] [answered]
Hi,

I'm trying to learn Aspire, and setup a dev environment on my mac where a new SQL server container is spun up.
Everything seems to be working.
My issue - I'd really like to connect to this database using Azure Data Studio as I debug to see data in there, but I can't seem to achieve that.

I use this in my apphost.

builder.Configuration["Parameters:sql-password"] = "P@$$w0rd1";
var sqlpassword = builder.AddParameter("sql-password", secret: true);
var sqldb = builder.AddSqlServer("sql", sqlpassword,port: 2022)
    .WithDataVolume()
    .AddDatabase("sqldb");

My assumption is that I should then be able to connect to this db while it's running in docker desktop using the below, however, I get the error **Login failed for user 'sa'** suggesting the password isn't being set to **P@$$w0rd1**
I have also tried using 127.0.0.1,2022 as I saw elsewhere localhost might be a problem, but not difference.
Anyone have any experience of this please?

![image](https://github.com/user-attachments/assets/8b25e57b-f404-4637-bbb8-63853820a029)

**Comment by davidfowl** [comment]
Localhost doesnt work for the sql server container, use 127.0.0.1

**Reply by Hypothesys-Smithgor** [reply]
Thanks for getting back to me David. As mentioned above, I tried 127.0.0.1 ,2022 as well and get the same error

**Reply by davidfowl** [reply]
https://www.twilio.com/en-us/blog/using-sql-server-on-macos-with-docker


Maybe a password strength problem.

PS: By default we generate strong passwords

**Reply by IEvangelist** [reply]
The problem is trying to itemize each part individually, instead just copy the entire connection string from the dashboard's corresponding resource detail. Once you have that, change the input type to Connection string and paste that in. See here: https://learn.microsoft.com/en-us/dotnet/aspire/database/sql-server-integration?tabs=dotnet-cli%2Cazure-data-studio#connect-to-database-resources

Please let me know if that helps!

**Comment by Hypothesys-Smithgor** [comment]
Just been checking a few different things, but this works:

docker run --name SQL2022 --platform=linux/amd64 \
           -e ACCEPT_EULA=1 \
           -e MSSQL_SA_PASSWORD=bigS4f3Password \
           -p 2022:1433 \
           -d mcr.microsoft.com/mssql/server:2022-latest

I can connect to this no problems.

However, when I run this via Aspire:

builder.Configuration["Parameters:sql-password"] = "bigS4f3Password";
var sqlpassword = builder.AddParameter("sql-password", secret: true);
var sqldb = builder.AddSqlServer("sql", sqlpassword,port: 2022)
    .WithDataVolume()
    .AddDatabase("sqldb");

It doesn't seem to work - sa login fails.
Another strange symptom is that the port doesn't show up in docker desktop when I load it via Aspire.

![image](https://github.com/user-attachments/assets/fecab1ce-6c8b-47c7-86a7-c67311ca6dc5)

I've upgraded the OS and taken latest docker desktop in the past few days - now wondering if something has broken there.

**Reply by davidfowl** [reply]
What version of aspire are you using?

**Comment by Hypothesys-Smithgor** [comment]
Latest I think?

![image](https://github.com/user-attachments/assets/1707ec1f-94f2-4826-bb4d-a03e1256156b)

definitely hitting the database too, as I see this in the log, so for some reason it looks like the password isn't being taken

![image](https://github.com/user-attachments/assets/239bb706-58b6-497a-a68d-06a02c8cea06)

**Reply by davidfowl** [reply, marked as answer]
If you delete the volume manually, does it work?

**Comment by Hypothesys-Smithgor** [comment]
Yes, it works!
Thank you so much (this has been frustrating me all day) and sorry to bother you on a Saturday.

Out of interest, localhost and 127.0.0.1 now both seem to work???

Also, docker shows a different port for the sql server, even though I'm able to connect on 2022.

![image](https://github.com/user-attachments/assets/6d4eb90c-3805-4118-b948-f62c21ba55c0)

![image](https://github.com/user-attachments/assets/1209d170-92d5-4974-b756-d05798227537)
